src/main/java/builder/dml/EntityColumn.java

```diff
@@ -0,0 +1,159 @@
+        Map<String, DMLColumnData> snapShotColumnMap = snapShotBuilderData.convertDMLColumnDataMap();
+
+        return this.columns.stream()
+                .filter(entityColumn -> {
+                    DMLColumnData persistenceColumn = snapShotColumnMap.get(entityColumn.getColumnName());
+                    return !entityColumn.getColumnValue().equals(persistenceColumn.getColumnValue());
+                })
+                .toList();
+    }
+
+    public void changeColumns(List<DMLColumnData> columns) {
+        this.columns = columns;
+    }
+
+    private <T> List<DMLColumnData> getInstanceColumnData(T entityInstance) {
+        Field[] fields = this.clazz.getDeclaredFields();
+        List<DMLColumnData> DMLColumnDataList = new ArrayList<>();
+        for (Field field : fields) {
+            getInstancePrimaryKey(DMLColumnDataList, field, entityInstance);
+            createDMLInstanceColumnData(DMLColumnDataList, field, entityInstance);
+        }
+        return DMLColumnDataList;
+    }
+
+    private <T> void getInstancePrimaryKey(List<DMLColumnData> DMLColumnDataList, Field field, T entityInstance) {
+        try {
+            if (field.isAnnotationPresent(Id.class)) {
+                field.setAccessible(true);
+                DMLColumnDataList.add(DMLColumnData.creatEntityPkColumn(field.getName(), field.getType(), field.get(entityInstance)));
+            }
+        } catch (IllegalAccessException e) {
+            throw new RuntimeException(GET_FIELD_VALUE_ERROR_MESSAGE + field.getName(), e);
+        }
+    }
+
+    private List<DMLColumnData> getEntityColumnData(Class<?> entityClass) {
+        Field[] fields = entityClass.getDeclaredFields();
+        List<DMLColumnData> DMLColumnDataList = new ArrayList<>();
+        for (Field field : fields) {
+            getEntityPrimaryKey(DMLColumnDataList, field);
+            createDMLEntityColumnData(DMLColumnDataList, field);
+        }
+        return DMLColumnDataList;
+    }
+
+    private void getEntityPrimaryKey(List<DMLColumnData> DMLColumnDataList, Field field) {
+        if (field.isAnnotationPresent(Id.class)) {
+            DMLColumnDataList.add(DMLColumnData.creatInstancePkColumn(field.getName(), field.getType()));
+        }
+    }
+
+    private void createDMLEntityColumnData(List<DMLColumnData> DMLColumnDataList, Field field) {
+        if (field.isAnnotationPresent(Transient.class) || field.isAnnotationPresent(Id.class) || field.isAnnotationPresent(OneToMany.class))
+            return; // @Transient인 경우 검증하지 않
+        // @Transient인 경우 검증하지 않음
+        String columnName = field.getName();
+
+        if (field.isAnnotationPresent(Column.class)) {
+            Column column = field.getAnnotation(Column.class);
+            columnName = column.name().isEmpty() ? columnName : column.name();
+        }
+
+        DMLColumnDataList.add(DMLColumnData.createEntityColumn(columnName));
+    }
```

Comment:
결과로 리턴하게 될 객체를 다음 메소드의 인자로 넘기면
어디서 결과값이 오염될 지 모르기 때문에 복잡한 시스템일 수록 지양하려고 하는 편이에요. 
함수형 프로그래밍의 패러다임을 이용하면 도움이 될 것 같습니다.

함수형으로 개선한다면 이런 느낌이 될 것 같습니다
```suggestion
private List<DMLColumnData> getEntityColumnData(Class<?> entityClass) {
        return Arrays.stream(entityClass.getDeclaredFields())
                .filter(field -> !field.isAnnotationPresent(Transient.class))
                .map(JoinEntityData::getDmlColumnData)
                .collect(Collectors.toList());
    }

    @NotNull
    private static DMLColumnData getDmlColumnData(Field field) {
        if (field.isAnnotationPresent(Id.class)) {
            return DMLColumnData.creatInstancePkColumn(field.getName(), field.getType());
        }
        return getColumnData(field);
    }

    @NotNull
    private static DMLColumnData getColumnData(Field field) {
        String columnName = field.getName();
        if (field.isAnnotationPresent(Column.class)) {
            Column column = field.getAnnotation(Column.class);
            columnName = column.name().isEmpty() ? columnName : column.name();
        }
        return DMLColumnData.createEntityColumn(columnName);
    }
```

Reply:
와 이렇게 깔끔하게 변할수있네요...!